src/UniversalViewer.ts

```diff
@@ -30,17 +30,24 @@ const ContentHandler: IContentHandlerRegistry = {
 };
 
 export class UniversalViewer extends BaseContentHandler<IUVData<any>> {
-  private _contentType: ContentType = ContentType.UNKNOWN;
-  private _assignedContentHandler: IContentHandler<IUVData<any>>;
+  public contentType: ContentType = ContentType.UNKNOWN;
+  public assignedContentHandler: IContentHandler<IUVData<any>>;
+
+  // for backwards compat, remove in next major version (UV5)
+  public _contentType = this.contentType;
+  public _assignedContentHandler;
+
   private _externalEventListeners: EventListener[] = [];
 
   constructor(public options: IUVOptions) {
     super(options);
     this._assignContentHandler(this.options.data);
+    // for backwards compat, remove in next major version (UV5)
+    this._assignedContentHandler = this.assignedContentHandler;
```

Comment:
One more question: since _assignContentHandler is async, would it be safer to make this assignment at the end of that method? Otherwise, unless I'm mistaken, we may try to copy the value before it has been assigned.

Reply:
yep, good point. have moved it